File: app/src/main/java/com/mgn/bingenovelreader/models/Novel.java
package com.mgn.bingenovelreader.models;

public class Novel {
    private long id;
    private String name;
    private String url;
    private String author;
    private String imageUrl;
    private Double rating;
    private String shortDescription;
    private String longDescription;
    private byte[] imageData;
    private String currentPageUrl;
    
    public long getId() {
        return this.id;
    }
    
    public void setId(long id) {
        this.id = id;
    }
    
    public String getName() {
        return this.name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public String getUrl() {
        return this.url;
    }
    
    public void setUrl(String url) {
        this.url = url;
    }
    
    public String getAuthor() {
        return this.author;
    }
    
    public void setAuthor(String author) {
        this.author = author;
    }
    
    public String getImageUrl() {
        return this.imageUrl;
    }
    
    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
    
    public Double getRating() {
        return this.rating;
    }
    
    public void setRating(Double rating) {
        this.rating = rating;
    }
    
    public String getShortDescription() {
        return this.shortDescription;
    }
    
    public void setShortDescription(String shortDescription) {
        this.shortDescription = shortDescription;
    }
    
    public String getLongDescription() {
        return this.longDescription;
    }
    
    public void setLongDescription(String longDescription) {
        this.longDescription = longDescription;
    }
    
    public byte[] getImageData() {
        return this.imageData;
    }
    
    public void setImageData(byte[] imageData) {
        this.imageData = imageData;
    }
    
    public String getCurrentPageUrl() {
        return this.currentPageUrl;
    }
    
    public void setCurrentPageUrl(String currentPageUrl) {
        this.currentPageUrl = currentPageUrl;
    }
    
}

File: app/src/main/java/com/mgn/bingenovelreader/models/Novel.kt
package com.mgn.bingenovelreader.models

class Novel {
    var id: Long = -1
    var name: String? = null
    var url: String? = null
    var author: String? = null
    var imageUrl: String? = null
    var rating: Double? = null
    var shortDescription: String? = null
    var longDescription: String? = null
    var imageFilePath: String? = null
    var currentPageUrl: String? = null
    var genres: List<String>? = null
    var favorite: Int = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val novel = o as Novel?

        return if (name != null) name == novel!!.name else novel!!.name == null
    }

    override fun hashCode(): Int {
        return if (name != null) name!!.hashCode() else 0
    }
}
